wasp/src/main/java/com/orhanobut/wasp/OkHttpStack.java

```diff
@@ -1,146 +1,159 @@
+import org.apache.http.message.BasicHeader;
+import org.apache.http.message.BasicHttpResponse;
+import org.apache.http.message.BasicStatusLine;
 
 import java.io.IOException;
-import java.io.InputStream;
-import java.net.CookieHandler;
-import java.net.HttpURLConnection;
-import java.net.URL;
-import java.security.KeyStore;
-import java.security.cert.CertificateException;
-import java.security.cert.X509Certificate;
-
-import javax.net.ssl.HostnameVerifier;
-import javax.net.ssl.SSLContext;
-import javax.net.ssl.SSLSession;
-import javax.net.ssl.SSLSocketFactory;
-import javax.net.ssl.TrustManager;
-import javax.net.ssl.TrustManagerFactory;
-import javax.net.ssl.X509TrustManager;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
 
 /**
  * @author Emmar Kardeslik
  */
-public class OkHttpStack extends HurlStack implements WaspHttpStack<HttpStack> {
+class OkHttpStack implements HttpStack {
 
-    private final OkUrlFactory okUrlFactory;
+    private OkHttpClient client;
 
-    public OkHttpStack() {
-        this(false);
+    OkHttpStack(final OkHttpClient client) {
+        this.client = client;
     }
 
-    public OkHttpStack(final OkHttpClient client) {
-        if (client == null) {
-            throw new NullPointerException("HttpClient may not be null.");
-        }
-        okUrlFactory = new OkUrlFactory(client);
+    OkHttpClient getClient() {
+        return client;
     }
 
-    public OkHttpStack(boolean trustAllCertificates) {
-        this(new OkHttpClient());
-        if (trustAllCertificates) {
-            setEmptyHostNameVerifier();
+    @Override
+    public HttpResponse performRequest(Request<?> request, Map<String, String> additionalHeaders)
+            throws IOException, AuthFailureError {
+
+        OkHttpClient okHttpClient = client.clone();
+
+        int timeoutMs = request.getTimeoutMs();
+        okHttpClient.setConnectTimeout(timeoutMs, TimeUnit.MILLISECONDS);
+        okHttpClient.setReadTimeout(timeoutMs, TimeUnit.MILLISECONDS);
+        okHttpClient.setWriteTimeout(timeoutMs, TimeUnit.MILLISECONDS);
+
+        com.squareup.okhttp.Request.Builder okHttpRequestBuilder = new com.squareup.okhttp.Request.Builder();
+        okHttpRequestBuilder.url(request.getUrl());
+
+        Map<String, String> headers = request.getHeaders();
+        for (final String name : headers.keySet()) {
+            okHttpRequestBuilder.addHeader(name, headers.get(name));
+        }
+        for (final String name : additionalHeaders.keySet()) {
+            okHttpRequestBuilder.addHeader(name, additionalHeaders.get(name));
         }
-    }
 
-    @Override
-    protected HttpURLConnection createConnection(URL url) throws IOException {
-        return okUrlFactory.open(url);
-    }
+        setConnectionParametersForRequest(okHttpRequestBuilder, request);
 
-    @Override
-    public HttpStack getHttpStack() {
-        return this;
-    }
+        com.squareup.okhttp.Request okHttpRequest = okHttpRequestBuilder.build();
```

Comment:
I think we can just use something like below instead of creating local variable. Thus we can avoid full package name
<pre>
Call okHttpCall = okHttpClient.newCall(okHttpRequestBuilder.build());
</pre>